### ISSUE TITLE
RealTime SadTalker

### ISSUE THREAD

**Issue by mrtpk** [Winfredy/SadTalker#806]
Any idea to make this real-time? I want to use this for a photo realistic talking head and integrate it with ChatGPT.  I want to use it for my head. So I have all the training videos and I plan to use it only with my image. Using these conditions can we eliminate / store computations that are ref photo dependant?  Thanks for any idea or pointers.

**Comment by UzairAsadBaig** [comment]
@mrtpk found any solution?

**Comment by mrtpk** [comment]
Not yet.

**Comment by nitinmukesh** [comment]
@mrtpk 
@UzairAsadBaig 

I came across 1 video which tries to explain realtime diffusion. See if it serves the purpose.
https://www.youtube.com/watch?v=WaNd9WT6yCI

**Comment by ghost** [comment]
@mrtpk Tell us if you find anything.

**Comment by lylepratt** [comment]
also interesting in real-time generation

**Comment by mrtpk** [comment]
I'm still searching for a real-time method. I will keep you posted. Thanks.

**Comment by mrtpk** [comment]
Geneplusplus got me about 20 FPS. Repo - https://github.com/yerfor/GeneFacePlusPlus

**Comment by Juand201** [comment]
Hi @mrtpk  did you achieve this with actual genefaceplusplus base code (im assuming you achieved to make it work real time right?) or you did some modifications to the code?

> Geneplusplus got me about 20 FPS. Repo - https://github.com/yerfor/GeneFacePlusPlus

**Comment by Abdelrhman-Wael** [comment]
@mrtpk how did you do it?

**Comment by mrtpk** [comment]
Hello,

Apologies for the late response.

The code in geneface++ is for audio with any length. If we set the audio length fixed, we can precompute many values that are needed for inference. This way we can have a great deal of speedup in inference.

Another thing is hubert model and other models (not sure of their names) were loaded each time an inference was made. This reloading was removed.

To make it streaming, queue was used for input. For the output ffmpeg was used.
